### ISSUE TITLE
Add camera tests for new Raspberry Pi cameras using libcamera/Picamera2

### ISSUE THREAD

**Issue by fernando79513** [canonical/checkbox#2842]
## Description

The current Raspberry Pi camera tests in Checkbox only support the legacy Raspberry Pi camera stack and effectively detect/test the Raspberry Pi Camera Module V2.

As investigated in #2588, the existing RPI camera detection relies on devices categorized as MMAL:

```
plugin: shell
category_id: com.canonical.plainbox::camera
id: camera/detect-rpi
estimated_duration: 1.0
imports: from com.canonical.plainbox import manifest
requires:
  manifest.has_rpi_camera == 'True'
command:
  udev_resource.py -f MMAL | grep "category: MMAL"
_summary: Detect presence of a MMAL camera.
```

The MMAL category is assigned when the device has vchiq as its subsystem:

```
# special device for PiCamera
if self._environment["SUBSYSTEM"] == "vchiq":
    return "MMAL"
```

This approach is tied to the legacy Raspberry Pi camera stack and does not detect newer Raspberry Pi cameras.

For Ubuntu 26.04 and newer Raspberry Pi platforms, camera support should use the modern libcamera stack instead.

Reference:
https://ubuntu.com/hardware/docs/boards/how-to/special_hardware/rpi-camera/

## Goal

Add new Checkbox camera tests capable of detecting and testing Raspberry Pi cameras supported through libcamera / Picamera2.

The new tests should coexist with the existing legacy RPI camera tests where necessary, rather than relying on the existing MMAL/vchiq detection mechanism.

## Proposed Implementation

A proof of concept using Picamera2 has already been verified successfully:

```python
from picamera2 import Picamera2
import cv2
import time

picam2 = Picamera2()

config = picam2.create_still_configuration(
    main={"size": (1920, 1080), "format": "RGB888"}
)

picam2.configure(config)
picam2.start()

# Give auto exposure / white balance a moment to settle
time.sleep(1)

frame = picam2.capture_array()

# Picamera2 gives RGB888 here; OpenCV expects BGR for saving
frame_bgr = cv2.cvtColor(frame, cv2.COLOR_RGB2BGR)

cv2.imwrite("capture.jpg", frame_bgr)

picam2.stop()

print("Saved capture.jpg")
```

This successfully captured an image on hardware where the existing MMAL-based test could not detect the camera.

## Dependencies

There are two options here

 - Include the required libcamera / Picamera2 dependencies in the Checkbox snap
 - Install the required dependencies through a Raspberry Pi-specific setup_include job.

The preferred approach should take into account snap confinement, image dependencies, and whether the test should work on a clean DUT without manually installing additional packages.

Follow-up to #2588, where it was determined that the missing RPI camera tests were caused by the current implementation only supporting the legacy Raspberry Pi camera stack.
